sdk/src/main/scala/com/horizen/account/state/ForgerStakeMsgProcessor.scala

```diff
@@ -366,37 +387,142 @@ case class ForgerStakeMsgProcessor(params: NetworkParams) extends FakeSmartContr
     stakeId
   }
 
+  private def getAllowedForgersIndexList(view: BaseAccountStateView): Array[Byte] = {
+
+    if (networkParams.allowedForgersList.isEmpty){
+      throw new IllegalStateException("Illegal call when list of forger is empty")
```

Comment:
why  do we have `IllegalStateException` exception type?
This call is used when someone tries to open forger stake, but there is nothing to open.
I expect the smart contract execution to `fail` but still be `valid`.

So, the type can be `ExecutionFailedException`(consume all gas) or `ExecutionRevertedException` (consume gas till this moment)

Reply:
Ok, I got it, in the "fake-evm" we check that allowedForgersList is not empty.
But I'm thinking can it be a source of "simple" mistake in the future?

Reply:
As an alternative we might move the code:

`if (networkParams.allowedForgersList.isEmpty)  throw new IllegalStateException`

directly into `isForgerListOpen()` and `getAllowedForgerListIndexes()` before the call to this method

Reply:
fine for now, thanks